File: diltakip-fe/src/interface/Student.ts
// Student interface remains the same
export interface Student {
    id: number;
    name: string;
    tckn: string;
    surname: string;
    phone: string;
    city: string;
    secondPhone?: string;
    address: string;
    registrationDate: string;
    updateTime?: string;
    createdOn: string;
}

// Page interface for paginated responses
export interface Page<T> {
    content: T[];
    totalPages: number;
    totalElements: number;
    size: number;
    number: number;
}

File: diltakip/src/main/java/com/dilokul/diltakip/model/entity/Student.java
package com.dilokul.diltakip.model.entity;

import com.dilokul.diltakip.enums.CityEnum;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "student")
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    private String name;
    private String tckn;
    private String surname;
    private String phone;

    @Column(name = "second_phone")
    private String secondPhone;
    private String address;

    @Enumerated(EnumType.STRING)
    private CityEnum city;

    @Column(name = "registration_date")
    private LocalDateTime registrationDate;

    @UpdateTimestamp
    @Column(name = "update_time")
    private LocalDateTime updateTime;

    @Column(name = "created_on", nullable = false, updatable = false)
    @CreationTimestamp
    private LocalDateTime createdOn;
}
